### DISCUSSION TITLE
How to utilize TimeTables to setup holidays?

### DISCUSSION THREAD

**Discussion by dillonmabry** [apache/airflow#25539] [Q&A]
I know in Autosys for example we can define a calendar setup, but has anyone actually implemented or has an example documentation on how to skip holidays via TimeTables? I read this documentation https://airflow.apache.org/docs/apache-airflow/stable/howto/timetable.html but the example didn't really cover holidays specifically. Any documentation or reading material would be appreciated.

**Comment by Taragolis** [comment]
I personally thought that documentation explain a lot of things about Timetable.

This link also might be helpful: https://www.astronomer.io/guides/scheduling-in-airflow/#example-custom-timetable
In additional you could have a look how implemented built-in Timetables: https://github.com/apache/airflow/tree/main/airflow/timetables

In the end EventsTimetable ([source](https://github.com/apache/airflow/blob/main/airflow/timetables/events.py), [doc](https://airflow.apache.org/docs/apache-airflow/stable/concepts/timetable.html#eventstimetable)) might be enough for your case.

And it would be nice if you provide more detail information about your requirements and approximate scheduling interval (hourly, daily, weekly and etc).

**Reply by dillonmabry** [reply]
> And it would be nice if you provide more detail information about your requirements and approximate scheduling interval (hourly, daily, weekly and etc).

For example, in banking, in US, we have federal holiday calendar. If I want to do some logic/processing based on if the day is a calendar holiday or not.

**Reply by Taragolis** [reply]
If it is not about skip holidays or run only on holidays and you actually need different pipeline in single dag than might be better use [PythonBranchOperator](https://github.com/apache/airflow/blob/55d0b9f45e06d1bc579653c12b6a7be680fb08fa/airflow/operators/python.py#L192) or create your own based on [BaseBranchOperator](https://github.com/apache/airflow/blob/main/airflow/operators/branch.py#L27-L54) (you could also have a look how it implemented in [BranchDayOfWeekOperator](https://github.com/apache/airflow/blob/main/airflow/operators/weekday.py))

Also additional links which might help you with branching
- https://airflow.apache.org/docs/apache-airflow/stable/concepts/dags.html?highlight=branching#branching
- https://www.astronomer.io/guides/airflow-branch-operator/
